### PR TITLE
Fix plugin related types for remarkable

### PR DESCRIPTION
**Pull request by pragdave**
If changing an existing definition:
- [ ] Provide a URL to documentation or source code which provides context for the suggested changes: <
See CHANGELOG.md.

Basically, the existing type definitions didn't extend to covering the plugin API

- [ ] Increase the version number in the header if appropriate.

n/a

- [ ] If you are making substantial changes, consider adding a `tslint.json` containing `{ "extends": "dtslint/dt.json" }`.

already there

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -21,7 +21,10 @@
         "lint": "dtslint types"
     },
     "devDependencies": {
+        "ava": "^0.25.0",
```
Comment (andy-ms) [inline package.json]: I don't think these new dependencies are needed?

File: types/remarkable/CHANGELOG.md [hunk 2]
```diff
@@ -0,0 +1,10 @@
+2018-10-07 Sun  <pragdave@gmail.com>
```
Comment (andy-ms) [inline types/remarkable/CHANGELOG.md]: We don't support CHANGELOG files -- these comments should go in the PR.

**Comment by pragdave** [comment]
2018-10-07 Sun  <pragdave@gmail.com>

  * The type `Rule` was being used for both parsing and rendering rules,
    but it was really just the rendering side. Rather than rename it and
    break things, I added a new type for ParsingRule and updated the
    `Rulers` function signatures.

  * The `content` field in `ContentToken` was typed as a `string`.
    However, its type is really `any`, as it is used simply to transport
    values between the parser and the renderer for any given rule.

File: package.json [hunk 3]
```diff
@@ -22,6 +22,7 @@
     },
     "devDependencies": {
         "dtslint": "github:Microsoft/dtslint#production",
+        "ts-node": "^7.0.1",
```
Comment (andy-ms) [inline package.json]: This also does not seem to be needed.

File: types/remarkable/lib/index.d.ts [hunk 4]
```diff
@@ -2,6 +2,7 @@ import * as Utils from "./common/utils";
 import Renderer = require("./renderer");
 import Ruler = require("./ruler");
 
+
```
Comment (andy-ms) [inline types/remarkable/lib/index.d.ts]: Extraneous whitespace change

**Pull request merged**